src/import-xml/import-xml.controller.ts

```diff
@@ -75,4 +76,14 @@ export class ImportXmlController {
     });
     return res.status(200).send(importedXmls);
   }
+
+  @ApiResponse({
+    status: 200,
+    description: 'Reprocesses the xml with the given id',
+  })
+  @Get(':id/reprocess')
+  async reprocessXML(@Res() res: Response, @Param('id') id: string) {
+    const importedXml = await this.importXmlService.reprocessXML(id);
```

Comment:
reprocessXML uses id as professorName, hence, it is not a good pratice to name it as "id" here. However, use a name to identify a person is not good too. There are many cases of researchers with the same name. Please change every place (including frontend) to use it as the real id.